src/Net/Http.fs

```diff
@@ -554,20 +554,39 @@ module private HttpHelpers =
         hasContentType.Value
 
     let getResponse (req:HttpWebRequest) silentHttpErrors =
-        if defaultArg silentHttpErrors false then
+
+        let getResponseAsync (req:HttpWebRequest) =
             async {
-                try
+                #if FX_NO_WEBREQUEST_TIMEOUT
                     return! Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
-                with
-                    | :? WebException as exn -> 
-                        if exn.Response <> null then 
-                           return exn.Response
-                        else 
-                            reraisePreserveStackTrace exn
-                            return Unchecked.defaultof<_>
+                #else
+                    let! child = 
+                        Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
+                        |> fun x -> Async.StartChild(x, req.Timeout)
```

Comment:
Also, would it be possible to somehow detect whether `req.Timeout` has been set and do the custom logic only when it is set? (And call the operation in a normal synchronous way otherwise.) I guess it could have some small overhead associated - not a big deal if this is hard to do though.